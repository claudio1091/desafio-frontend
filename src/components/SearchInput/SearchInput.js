import React from 'react';
import styled from 'styled-components';
import Theme from '../../config/Theme';
import icon from '../../assets/icons/search.svg';

function SearchInput(props) {
  const SearchContainer = styled.div`
    display: flex;
    padding: 0.7em;
    background: ${Theme.colors.background};
    border: 1px solid ${Theme.colors.black_38};
    width: 30em;

    @media (max-width: 700px) {
      width: 20em;
    }

    -webkit-box-shadow: 0px 3px 8px 0px ${Theme.colors.boxShadow};
    -moz-box-shadow: 0px 3px 8px 0px ${Theme.colors.boxShadow};
    box-shadow: 0px 3px 8px 0px ${Theme.colors.boxShadow};
  `;

  const SearchInput = styled.input`
    font: inherit;
    background: transparent;
    border: none;
    width: 100%;

    &:focus {
      outline: none;
    }
  `;

  const SearchIcon = styled.img`
    height: ${Theme.font.body.fontSize};
    margin-left: ${Theme.font.body.fontSize};
    color: ${Theme.colors.black_38};
  `;

  const inputPlaceholder = 'Pesquise por uma informação do processo';

  return (
    <SearchContainer>
      <SearchInput type="text" placeholder={inputPlaceholder} value={props.value} onChange={props.onChange} />
      <SearchIcon src={icon} alt="logo" />
    </SearchContainer>
  );
}

export default SearchInput;
