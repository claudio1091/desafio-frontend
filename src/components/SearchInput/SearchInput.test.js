import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';
import SearchInput from './SearchInput';

configure({ adapter: new Adapter() });

describe('Componente SearchInput', () => {
  it('renderizar sem crash', () => {
    const tree = shallow(<SearchInput />);
    expect(tree).toMatchSnapshot();
  });

  it('renderizar com parâmetro', () => {
    const searchValue = 'Alguma informação a pesquisar';
    const wrap = mount(<SearchInput value={searchValue} />);
    expect(wrap.find('input').prop('value')).toEqual(searchValue);
  });
});
