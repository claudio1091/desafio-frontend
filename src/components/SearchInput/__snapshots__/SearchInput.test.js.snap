// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente SearchInput renderizar sem crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.input
          placeholder="Pesquise por uma informação do processo"
          type="text"
          value={undefined}
/>,
        <styled.img
          alt="logo"
          src="search.svg"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "placeholder": "Pesquise por uma informação do processo",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "logo",
          "src": "search.svg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.input
            placeholder="Pesquise por uma informação do processo"
            type="text"
            value={undefined}
/>,
          <styled.img
            alt="logo"
            src="search.svg"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "placeholder": "Pesquise por uma informação do processo",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "logo",
            "src": "search.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
